pkg/ddevapp/ddevapp.go

```diff
@@ -1594,31 +1594,52 @@ func FindNotOmittedImages(app *DdevApp) []string {
 	return images
 }
 
-// FindMaxTimeout looks through all services and returns the max timeout found
-// Defaults to 120
-func (app *DdevApp) FindMaxTimeout() int {
-	const defaultContainerTimeout = 120
-	maxTimeout := defaultContainerTimeout
+// GetMaxContainerWaitTime looks through all services and returns the max time we expect
+// to wait for all containers to become `healthy`. Mostly this is healtcheck.start_period.
+// Defaults to DefaultContainerTimeout (usually 120 unless overridden)
+func (app *DdevApp) GetMaxContainerWaitTime() int {
+	defaultContainerTimeout, _ := strconv.Atoi(app.DefaultContainerTimeout)
+	maxWaitTime := defaultContainerTimeout
+
 	if app.ComposeYaml == nil {
 		return defaultContainerTimeout
 	}
 	if y, ok := app.ComposeYaml["services"]; ok {
 		for _, v := range y.(map[string]interface{}) {
 			if i, ok := v.(map[string]interface{})["healthcheck"]; ok {
-				if timeout, ok := i.(map[string]interface{})["timeout"]; ok {
-					duration, err := time.ParseDuration(timeout.(string))
+				if startPeriod, ok := i.(map[string]interface{})["start_period"]; ok {
+					duration, err := time.ParseDuration(startPeriod.(string))
 					if err != nil {
 						continue
 					}
 					t := int(duration.Seconds())
-					if t > maxTimeout {
-						maxTimeout = t
+					if t > maxWaitTime {
+						maxWaitTime = t
+					}
+				} else { /* In this case we didn't have a specified start_period, so guess at one */
+					interval := 1
+					retries := 3
+					maxWaitTime = 15 /* Use a generic default for start_period/maxWaitTime based on 3*5 */
```

Comment:
The comment is confusing.
I don't understand where `15` comes from, `3*5`, if `3` is retries, than what is `5` ?
Does it come from `600/120` ? (600 is the value I see in `snapshot.go`.)
In this case the comment about `start_period/maxWaitTime` should be `maxWaitTime/start_period`.

Reply:
5 is the default `start_period`. If the `test` were to fail 3 times, it would be retried in 5 seconds, meaning 15 seconds or more. I'll see if I can add a comment there.